client/lib/url/decode-utils.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import { URL as TypedURL } from 'types';
+
+interface Stringable {
+	toString: () => string;
+}
+
+/**
+ * Wrap decodeURI in a try / catch block to prevent `URIError` on invalid input
+ * Passing a non-string value will return an empty string.
+ * @param  encodedURI URI to attempt to decode
+ * @return            Decoded URI (or passed in value on error)
+ */
+export function decodeURIIfValid( encodedURI: string | Stringable ): TypedURL {
```

Comment:
It's really hard to make sense of this function in a typed world 🙃 

It seems to be a helper so `decodeURI` is safe and won't crash the app. In that sense, it should probably accept and argument of type `Stringable | Falsy`.

```suggestion
export function decodeURIIfValid( encodedURI: Stringable | Falsy ): TypedURL {
```

This type of refactor would be much simpler if we had type safety, but that's a long ways away. At the moment, we'll need to open up the argument type and be more careful in the body. Callers are not updated, so we can continue to expect "bad" inputs.

Opening up the argument type requires a few other changes…